beacon-chain/core/blocks/block.go

```diff
@@ -0,0 +1,77 @@
+package blocks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/prysmaticlabs/prysm/beacon-chain/utils"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+var clock utils.Clock = &utils.RealClock{}
+
+// NewGenesisBlock returns the canonical, genesis block for the beacon chain protocol.
+func NewGenesisBlock(stateRoot []byte) *pb.BeaconBlock {
+	block := &pb.BeaconBlock{
+		ParentRootHash32:              []byte{0},
+		RandaoRevealHash32:            []byte{0},
+		CandidatePowReceiptRootHash32: []byte{0},
+		StateRootHash32:               []byte{0},
+	}
+	// Genesis time here is static so error can be safely ignored.
+	// #nosec G104
+	protoGenesis, _ := ptypes.TimestampProto(params.BeaconConfig().GenesisTime)
+	block.Timestamp = protoGenesis
+	block.StateRootHash32 = stateRoot
+	return block
+}
+
+// IsRandaoValid verifies the validity of randao from block by comparing it with
+// the proposer's randao from the beacon state.
+func IsRandaoValid(blockRandao []byte, stateRandao []byte) bool {
+	var h [32]byte
+	copy(h[:], stateRandao)
+	return hashutil.Hash(blockRandao) == h
+}
+
+// IsSlotValid compares the slot to the system clock to determine if the block is valid.
```

Comment:
Can you expand on this for godocs? 

Something like: A slot is considered valid if the current time within the slot duration from genesis.